### DISCUSSION TITLE
Tiktok/Youtube Shorts preset(s)?

### DISCUSSION THREAD

**Discussion by LearningJer** [SubtitleEdit/subtitleedit#7370] [Ideas]
Would it be possible to add a tiktok/vertical profile preset? 

For a lot of the videos I work on the people are talking fast so having a short text limit (10-20 characters per line) and 1-2 lines would be perfect.  

Here is an example I just found (ignore the caption formatting/animations)
https://youtube.com/shorts/MIw_qotCXcQ?si=qRS1t0atOnwgtx7n

**Comment by LeonCheung** [comment]
Totally agree. I was about to propose it a long time ago. XD

![vertical](https://github.com/SubtitleEdit/subtitleedit/assets/494788/a1ef5146-8a72-4562-976d-d168d8e8ebee)

More than that, it could be better to have multiple profiles for different needs. Say, to swap the video player to the left side, because people have different habits. And it's especially convenient when working with multiple monitors (e.g. when an external text editor is on the secondary monitor which is on the right side of the desk, so that you don't have to have the mouse traveling a longer distance). Another benefit of such a layout is that now we have the List, Textbox, and Waveform well aligned vertically, which is more user-friendly.  So maybe a separate "Layout" menu entry is a better idea:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/efa6ebc0-97e2-42f6-815b-759740a02e16)

**Comment by LearningJer** [comment]
I was more talking about the profiles under Settings to have the subtitles formatted for vertical timelines. I had tried messing with it to format the subtitles similar to the video in the OP.

![image](https://github.com/SubtitleEdit/subtitleedit/assets/143230131/78241c5d-d7e7-4b1c-b926-83f3bea75762)

I actually like what you mentioned as well though having a tab for the layouts for the different setups/workflows would be really useful! :)

**Reply by LeonCheung** [reply]
Ah, right. Indeed, there are more and more videos with vertical dimensions generated. It's time to have a better workflow for that, one way or another.

**Comment by niksedk** [comment]
How is latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

New profile: 
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/b3da77f8-2e37-44d0-b1e2-ee282b16a56a)

New layout:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/de122528-e062-4f73-8c66-1b7bbcd4e447)

You can toggle layout via the shortcut: Ctrl+Shift+Alt+L 
Note: this will require some changes in the SE UI, the toolbar icons "Show/hide video" and "Show/hide waveform" will be history... and we will have to add a new icon for "Layout".

@jmaraujouy: Do you have time for a "Layout" toolbar image?
(SE will probably have around 7-9 layouts)

**Reply by LearningJer** [reply]
Ooooh excited to try this out!

**Reply by LeonCheung** [reply]
Amazing work! That's a really really cool feature.
@niksedk, Before the formal icons come out, you may want something for testing. Just drew a few for you. Hope it helps.
[Layout_icons_temp.zip](https://github.com/SubtitleEdit/subtitleedit/files/12640851/Layout_icons_temp.zip)
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/c704f2a1-b272-480a-9347-f8391f192956)

Just curious what kind of interface would pop up when clicking it...

**Comment by ivandrofly** [comment]
@niksedk about the new icons, do I need to do something 
in order to enable them?

In installed version of SE I can see them but in debug I cannot! It still shows the old icons even though
my branch is in sync with current "main"

**Comment by niksedk** [comment]
Thx for the icons - included here: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

New icons included in above beta (and latest main)

**Reply by LeonCheung** [reply]
Nice solution.  :)

One tiny report:

When trying to have the combination as below, (I know it's not a good idea):
![s2](https://github.com/SubtitleEdit/subtitleedit/assets/494788/56d975b0-b4bc-4922-90e5-15590046f203)

it seems not in the appropriate version:
![s1](https://github.com/SubtitleEdit/subtitleedit/assets/494788/2bda7e2f-6917-41b8-bda2-4ab555a257c2)

-------
UPDATE:

Not sure why but it's gone with @jmaraujouy 's version. Please ignore it. :)

**Comment by jmaraujouy** [comment]
> @jmaraujouy: Do you have time for a "Layout" toolbar image? (SE will probably have around 7-9 layouts)

@niksedk 
I was working on a new icon, but I see @LeonCheung already took care of the issue.

But you can use the Legacy icon that is included in the zip file.

![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/833babe5-bf43-4890-aca2-2005209477f8)

[Icons.zip](https://github.com/SubtitleEdit/subtitleedit/files/12641167/Icons.zip)

![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/09076bea-73b0-495e-878d-ade163fc9f9d)

**Reply by LeonCheung** [reply]
Mine are just very rough versions for temp testing XD.

**Reply by jmaraujouy** [reply]
No, I like them. That's why I keep reworking them to look like yours (I posted them in the message right below).

My initial design seems a little "crowded".

**Comment by jmaraujouy** [comment]
And here's Leon's version, but rotated 180º to fit better with the SE layout default design, and with fixed colors.

![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/1ca4afc8-6ce2-48c8-a5fc-66f03ebe6c16)

[Icons2.zip](https://github.com/SubtitleEdit/subtitleedit/files/12641195/Icons2.zip)

I also played a little with size and position, so it matches the _Beautify Time Codes_ icon.

**Comment by LeonCheung** [comment]
@niksedk

1. I was testing every possible edge case, such as the **un-dock video** feasure (maybe it's on your TODO list already).

2. Please consider adding a toggle for this on the pop-up window:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/b9d17cb1-7e08-43c7-97f1-df2e1639f00a)

Because obviously it doesn't quite fit some profiles especially for those who seldom need that panel. Besides, I believe this feature is so "hidden" to many users. So this is a good chance to expose it better. :) Maybe like this:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/e305f84b-5d20-4acb-8f53-f25583da4cb0)

3. If possible, it would be nice to auto-adjust the aspect ratio of the player based on the source video (when switching among profiles) to wipe out those "black bands", so that we can leverage the UI much better and save some UI re-tweaking work.

**Comment by niksedk** [comment]
@jmaraujouy: thx for icons, included in latest beta :)

@LeonCheung: Many fixes for layout + shortcut... some auto size (but no video AR yet...)

Beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/69640ff0-1aae-4f03-9c40-cf199c2125c0)
(also keyboard support in the layout window + numbering)

**Reply by LeonCheung** [reply]
1. Auto-size works nice! Maybe L7 should be tweaked as well because the textbox seems too narrow even with vertical videos (and it IS mainly for vertical videos, right?)
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/2e68601e-eebd-4b9a-8b32-3036d5165b08)
2. After un-docking, I highly suggest that the layout feature should be temporally disabled until re-docking... I just noticed things messed up otherwise...
3. It seems that L2, L5 and L6 img should be moved down about 1px. for better alignment (not so sure because the offset is tiny).
4. I've tried to improve the graphics for them, replace if you think it's a better version. [layout chooser img refined.zip](https://github.com/SubtitleEdit/subtitleedit/files/12643065/layout.chooser.img.refined.zip) 
or: [layout chooser img refined - v2.zip](https://github.com/SubtitleEdit/subtitleedit/files/12643151/layout.chooser.img.refined.-.v2.zip) :)


5. Another issue is about the response/performance for a long video/subtitle. Testing with a 3k-line subtitle (no video loaded), it takes about 2.5 to 3 secs here on my PC. After loading the video source, it takes about 3 to 4 secs.
6. Maybe a new toggle to add to Settings > Toolbar.

**Comment by niksedk** [comment]
1) I think perhaps some video control toolbar should exist for layouts with video but without waveform... like a horizontal version of the current video controls....

2) He, never thought of that... disabled a little...

3) Yes, done.

4) Nice, included - the old gfx was actually ripped from you ;)

5) omg, that's slow. Apparently that's how moving controls are... 
Hack applied: if more than 1000 lines, SE will clear list view, transition layout, fill list view.
I think the list view support something called "virtual mode"... will require a lot of rewriting, but that could be the way to go

6) Toggle for what?

Beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

**Reply by LeonCheung** [reply]
1. No problem. :) 
2. Maybe hide the button as well (or displaying a tooltip like "Cannot use it while un-docking")... Currently, users can still see and click it, and may get confused by seeing nothing happening.
3. Much better 👍
4. Hehe I know, but hey, feel free to, anytime. XD
5. It is faster indeed, not sure how but really great!
6. I mean the visibility toggle for the "Layout" button, considering we have toggles for all other toolbar icons in Settings. (just in case you forgot to add)

---
Surprise:

With Layout 5/6/8, now we can finally work in Portrait orientation mode with SE on the 2nd monitor, yay!

![img](https://github.com/SubtitleEdit/subtitleedit/assets/494788/ff0989c5-fb88-40f0-ab2b-60528f26c41e)

**Reply by LeonCheung** [reply]
One more thing, For Translation mode, the textbox width may need an update/refresh after a layout change:

A GIF:
![GIF 9-18-2023 3-27-36 PM](https://github.com/SubtitleEdit/subtitleedit/assets/494788/deebccb7-b41c-4d4b-80ca-e052fc6fa52f)

Plus an error message when trying to quit translation mode by closing one side of the subtitle:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/3ae6caeb-54bd-4e97-8fc9-480308ea95a2)

**Reply by abc16361** [reply]
Can the selected items on the screenshot be put together on one side, left or right, to make the textbox wider? Putting them together may make it look more organized. What do you think?
<img width="1094" alt="Snipaste_2309181949" src="https://github.com/SubtitleEdit/subtitleedit/assets/101362122/dec9e448-1f42-4e65-b9ae-c6e9a2b404cd">

**Reply by LeonCheung** [reply]
@abc16361 I think it should be discussed in a separate topic. See #7422
